modules/gcp/cac-igm/cac-startup.sh.tmpl

```diff
@@ -38,156 +40,189 @@ get_credentials() {
-    fi
-    TIMEOUT=$((TIMEOUT-10))
-    sleep 10
-done
-
-# Check that the domain name can be resolved and that the LDAP port is accepting
-# connections. This could have been all done with the ldapwhoami command, but
-# due to a number of occasional cac-installation issues, such as "domain
-# controller unreachable" or "DNS error occurred" errors, check these explicitly
-# for logging and debug purposes.
-
-echo '### Ensure domain ${domain_name} can be resolved ###'
-TIMEOUT=1200
-until host ${domain_name}; do
-    if [ $TIMEOUT -le 0 ]; then
-        break
-    else
-        echo "Trying to resolve ${domain_name}. Retrying in 10 seconds... (Timeout in $TIMEOUT seconds)"
-    fi
-    TIMEOUT=$((TIMEOUT-10))
-    sleep 10
-done
-
-echo '### Ensure domain ${domain_name} port 636 is reacheable ###'
-TIMEOUT=1200
-until netcat -vz ${domain_name} 636; do
-    if [ $TIMEOUT -le 0 ]; then
-        break
+wait_for_dc() {
+    log "--> Ensure AD account is available..."
+    TIMEOUT=1200
+    until ldapwhoami \
+        -H ldap://${domain_controller_ip} \
+        -D ${ad_service_account_username}@${domain_name} \
+        -w $AD_SERVICE_ACCOUNT_PASSWORD \
+        -o nettimeout=1; do
+        if [ $TIMEOUT -le 0 ]; then
+            break
+        else
+            log "--> Waiting for AD account ${ad_service_account_username}@${domain_name} to become available. Retrying in 10 seconds... (Timeout in $TIMEOUT seconds)"
+        fi
+        TIMEOUT=$((TIMEOUT-10))
+        sleep 10
+    done
+
+    # Check that the domain name can be resolved and that the LDAP port is accepting
+    # connections. This could have been all done with the ldapwhoami command, but
+    # due to a number of occasional cac-installation issues, such as "domain
+    # controller unreachable" or "DNS error occurred" errors, check these explicitly
+    # for logging and debug purposes.
+    log "--> Ensure domain ${domain_name} can be resolved..."
+    TIMEOUT=1200
+    until host ${domain_name}; do
+        if [ $TIMEOUT -le 0 ]; then
+            break
+        else
+            log "Trying to resolve ${domain_name}. Retrying in 10 seconds... (Timeout in $TIMEOUT seconds)"
+        fi
+        TIMEOUT=$((TIMEOUT-10))
+        sleep 10
+    done
+
+    log "--> Ensure domain ${domain_name} port 636 is reacheable..."
+    TIMEOUT=1200
+    until netcat -vz ${domain_name} 636; do
+        if [ $TIMEOUT -le 0 ]; then
+            break
+        else
+            log "Trying to contact ${domain_name}:636. Retrying in 10 seconds... (Timeout in $TIMEOUT seconds)"
+        fi
+        TIMEOUT=$((TIMEOUT-10))
+        sleep 10
+    done
+}
+
+install_cac() {
+    log "--> Installing Cloud Access Connector..."
+    export CAM_BASE_URI=${cam_url}
+
+    log "--> Running command: $INSTALL_DIR/cloud-access-connector install"
+    log "--> CAC install options:"
+    log "-t $${CAC_TOKEN} --accept-policies"
```

Comment:
since we treat cac token and pcoip reg code as secrets, we shouldn't print them out.